### ISSUE TITLE
hash-ref with hashtable keys

### ISSUE THREAD

**Issue by cderici** [pycket/pycket#239]
Finally, here's what's going on:

\> Racket
\> (hash-ref (make-hash '((#hasheq() . 1))) #hasheq() #f)
1
\> (hash-ref (make-hash '((#hasheq((a . 3)) . 1))) #hasheq((a . 3)) #f)
1

\> Pycket
\> (hash-ref (make-hash '((#hasheq() . 1))) #hasheq() #f)
1
\> (hash-ref (make-hash '((#hasheq((a . 3)) . 1))) #hasheq((a . 3)) #f)
#f

**Comment by samth** [comment]
When I run those on pycket, they both produce `#f` and both of them seem to generate equal hashes in the reader.

**Comment by samth** [comment]
however 
```
> (hash-ref (make-hash (list (cons (hasheq) 1))) (hasheq) #f) 
1
```

**Comment by samth** [comment]
Well here's the real problem:

```
> (equal? (hasheq (cons 1 2) 1) (hasheq (cons 1 2) 1))
#f
```

**Comment by samth** [comment]
```
> (hash-ref (make-hash (list (cons (hash) 1))) (hash) #f) 
[1538751617.230688] compiling linklet : (linklet ((.top-level-bind! .top-level-require!) (.mpi-vector .syntax-literals) (.namespace .phase .self .inspector .bulk-binding-registry .set-transformer!)) () (hash-ref (make-hash (list (cons (hash) 1))) (hash) #f))
[1538751617.231057] making instance : link
[1538751617.231121] making instance : instance
#f
``

**Comment by krono** [comment]
Am 5. Oktober 2018 22:18:53 MESZ schrieb Caner Derici <notifications@github.com>:
>Closed #239 via e757a3271f3aecc49080496278e8ece890b4a1ac.

i still love the doctests :D
-- 
Sent from mobile device
